kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventListener.java

```diff
@@ -0,0 +1,70 @@
+package org.hypertrace.core.kafka.event.listener;
+
+import com.typesafe.config.Config;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.BiConsumer;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.serialization.Deserializer;
+
+/**
+ * KafkaEventListener consumes events from a single Kafka Topic and on every message invokes
+ * provided callbacks. The callback invocation is done in a separate thread and needs to be
+ * concurrent safe.
+ */
+public class KafkaEventListener<K, V> implements AutoCloseable {
+  private final KafkaEventListenerThread<K, V> kafkaEventListenerThread;
+
+  private KafkaEventListener(KafkaEventListenerThread<K, V> kafkaEventListenerThread) {
+    this.kafkaEventListenerThread = kafkaEventListenerThread;
+    kafkaEventListenerThread.start();
+  }
+
+  @Override
+  public void close() throws Exception {
+    kafkaEventListenerThread.interrupt();
```

Comment:
Note you probably want to shutdown the whole executor on closed. Doing so should also shutdown the threads.